### DISCUSSION TITLE
Ikea Remote 4 button, only creates 1 entity & event fired for left and right are the same event

### DISCUSSION THREAD

**Discussion by magiva** [arendst/Tasmota#14700] [Ideas]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
Home assistant entity created for the IKEA N2 4 button remote, creates 1 entity, none for the buttons.
If i monitor the event, the events for left and right button are identicial and cannot be separated for actions

Right press
![image](https://user-images.githubusercontent.com/7522405/152197698-ff14890b-2c01-4083-a7e5-5b5f8c8f8782.png)

Left press
![image](https://user-images.githubusercontent.com/7522405/152197910-0cbe814a-ac71-4ca7-b855-9e4c19265b58.png)

Initially i thought i could use the args array as i see 256 for right and 257 for left but these values are gone if the button is held.
so, technically single press can be identitied but a held press cannot.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
Are you running it in ZHA mode? In that case, the details of operation has *nothing* to do with Tasmota, which simply provides a ser2net tunnel from ZHA to the Zigbee chip.

**Comment by magiva** [comment]
ok, whilst i understand that i have 2 questions, 
1, how do i swap modes from zha mode ?  
2, if its tunnelling the information directly then why is it the same for left and right buttons?, seems that there is piece of data missing or the label of "press" needed the words left and right adding, where would that issue be logged so it can be addressed.

**Reply by sfromis** [reply]
2 - Because ZHA handles *everything*. Whatever you have of issues around ZHA handling of devices is in no way whatsoever a Tasmota question. Ask them, not us. ZHA is a feature of Home Assistant.

1 - If you config `Module 75` again, you'll get Z2T mode (with MQTT payloads) instead of ZHA. Note: This works very differently.

**Comment by magiva** [comment]
I'm more familiar with mqtt so I'll do that. i would have liked to have
raised the issue with the right group as it won't get fixed otherwise

appreciate the answer

**Reply by sfromis** [reply]
Raising the issue with the right group is raising it with Home Assistant. Tasmota is just not involved with the details when it is ZHA mode. I don't know if that specific device works with Z2T/Tasmota, but if not, you are at the right place :-)

**Comment by magiva** [comment]
done, you can close this ticket
